File: easy/cpp/Question643.cpp
#include <algorithm>
#include <climits>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    double findMaxAverage(vector<int> &nums, int k) {
        int ans = INT_MIN, sum = 0;
        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];
            if (i < k - 1) {
                continue;
            }
            ans = max(ans, sum);
            sum -= nums[i - k + 1];
        }
        return 1. * ans / k;
    }
};

int main() {
    vector<int> nums = {1, 12, -5, -6, 50, 3};
    cout << Solution().findMaxAverage(nums, 4) << endl;
    return 0;
}

File: easy/python/Question643.py
from typing import List


class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        # 这里的ans和aver其实是总和，起变量名要见名知意，检讨
        ans, aver = float("-inf"), 0
        for i in range(len(nums)):
            # right = i
            # right - left + 1 = k
            # left = i - k + 1
            aver += nums[i]
            if i < k - 1:
                continue
            else:
                ans = max(ans, aver)
                aver -= nums[i - k + 1]
        return ans / k


print(Solution().findMaxAverage([-1], 1))
